testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -2674,35 +2661,14 @@ fn stx_transfers_dont_effect_idle_timeout() {
     // Add a delay to the block validation process
     TEST_VALIDATE_DELAY_DURATION_SECS.lock().unwrap().replace(5);
 
-    let info_before = signer_test
-        .stacks_client
-        .get_peer_info()
-        .expect("Failed to get peer info");
-    let blocks_before = signer_test
-        .running_nodes
-        .nakamoto_blocks_mined
-        .load(Ordering::SeqCst);
+    let info_before = signer_test.get_peer_info();
+    let blocks_before = signer_test.running_nodes.nakamoto_blocks_mined.get();
     info!("---- Nakamoto booted, starting test ----";
         "info_height" => info_before.stacks_tip_height,
         "blocks_before" => blocks_before,
     );
     signer_test.mine_nakamoto_block(Duration::from_secs(30));
 
-    info!("---- Verifying that the block was mined ----");
-    // Verify that the block was mined
-    wait_for(30, || {
-        let blocks_mined = signer_test
-            .running_nodes
-            .nakamoto_blocks_mined
-            .load(Ordering::SeqCst);
-        let info = signer_test
-            .stacks_client
-            .get_peer_info()
-            .expect("Failed to get peer info");
-        Ok(blocks_mined > blocks_before && info.stacks_tip_height > info_before.stacks_tip_height)
```

Comment:
Are you sure that removing the check on `nakamoto_blocks_mined` does not cause any flakiness?

Reply:
I'm not totally sure, but I've added the `nakamoto_blocks_mined` check to `mine_nakamoto_block` to be extra careful. That function also has the `info_before` check.